python/ray/serve/_private/router.py

```diff
@@ -483,7 +483,7 @@ async def assign_request(
                     )
                     if isinstance(ref, ray.ObjectRef):
                         ref._on_completed(callback)
-                    else:
+                    elif isinstance(ref, ray.ObjectRefGenerator):
```

Comment:
why's this needed?

Reply:
Sorry should have called out please don't merge yet, I still need to fix this, it's because `FakeObjectRef` and such don't have fake implementations of `.completed`.

Reply:
oops, we should it to the fakes